docs/content/guides/dagster/re-execution.mdx

```diff
@@ -0,0 +1,49 @@
+---
+title: Re-execution | Dagster
+---
+
+# Re-execution in Dagster
+
+This guide is intended to walk through how to re-execute Dagster pipelines and where the subsequent executions are found within Dagit.
+
+## Motivation
+
+If solids fail or upstream data has changed within a pipeline execution, the pipeline may need to be re-run starting from a particular point. We call this process `re-execution`. 
+
+Imagine a machine learning pipeline with three solids, the first being the most time and resource intensive: training the model, testing the model, and building analyses on the results. If the pipeline fails with the solid that is testing the model, after fixing the root cause it would take much more time to create a new run of the pipeline than to start again from the second step.
+
+With Dagster, the re-execution of parts of the pipeline is grouped with the original run to make it easy to trace. The original pipeline execution metadata is not overwritten, making re-execution a non-destructive operation.
+
+## Example Walk-through
+
+Upon navigating to a particular run or launching an execution within Dagit, the re-execution option is found on the top right.
```

Comment:
Can we write an example pipeline to let readers load and follow the guide along with? 

High-level speaking, we'd like a guide to be a step-by-step practical exercise that readers can follow through and learn how to achieve a goal.

Reply:
I created an example pipeline, `unreliable_pipeline`. This takes the "unreliable" idea (random failure 50% of the time) as well as an IO manager to show that re-execution support inputs and outputs passed between tasks.

Is the `examples/reexecution` folder the appropriate place to put this?